diff --git a/net/wireless/util.c b/net/wireless/util.c
index 534b7ed..b283acb 100644
--- a/net/wireless/util.c
+++ b/net/wireless/util.c
@@ -641,10 +641,80 @@ void ieee80211_amsdu_to_8023s(struct sk_buff *skb, struct sk_buff_head *list,
 }
 EXPORT_SYMBOL(ieee80211_amsdu_to_8023s);
 
+/* VI, VO, BE, BK */
+
+static const u8 dscp_table[] = {
+	0, /* BE = BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	1, /* BK */
+	0, /* BE */
+	2, /* BK */
+	0, /* BE */
+	2, /* BK */
+	0, /* BE */
+	2, /* BK */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	3, /* AF21 */
+	0, /* BE */
+	3, /* AF22 */
+	0, /* BE */
+	3, /* AF23 */
+	0, /* BE */
+	4, /* AF31 */
+	0, /* BE */
+	4, /* AF32 */
+	0, /* BE */
+	4, /* AF33 */
+	0, /* BE */
+	4, /* CS4 */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	4, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	7, /* EF = VOICE */
+	0, /* BE */
+	5, /* CS6 = VI  */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	5, /* CS7 = IC */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0, /* BE */
+	0  /* BE */
+	};
+
+  
 /* Given a data frame determine the 802.1p/1d tag to use. */
 unsigned int cfg80211_classify8021d(struct sk_buff *skb)
 {
-	unsigned int dscp;
+ 	unsigned int dscp;	  
 
 	/* skb->priority values from 256->263 are magic values to
 	 * directly indicate a specific 802.1d priority.  This is used
@@ -661,8 +731,8 @@ unsigned int cfg80211_classify8021d(struct sk_buff *skb)
 	default:
 		return 0;
 	}
+	return (dscp_table[dscp>>2]);
 
-	return dscp >> 5;
 }
 EXPORT_SYMBOL(cfg80211_classify8021d);
 
